lib/project_types/script/forms/create.rb

```diff
@@ -20,7 +20,11 @@ def ask_extension_point
       end
 
       def ask_name
-        CLI::UI::Prompt.ask(@ctx.message('script.forms.create.script_name'))
+        name = CLI::UI::Prompt.ask(@ctx.message('script.forms.create.script_name'))
+        unless name.match?(/[^0-9A-Za-z _-]/)
```

Comment:
Nitpick: Stylistically I usually prefer to put small conditionals on a single line. ie `return foo if bar`